plugins/reach/plugins/OpenCalais/batch/KReachVendorTaskOpenCalaisProcessorEngine.php

```diff
@@ -0,0 +1,777 @@
+<?php
+
+require_once (__DIR__ . DIRECTORY_SEPARATOR . 'Constants.php');
+require_once (__DIR__ . DIRECTORY_SEPARATOR . 'RuleEngine.php');
+
+/**
+ * @package plugins.openCalaisReachVendor
+ * @subpackage batch
+ */
+class KReachVendorTaskOpenCalaisProcessorEngine extends KReachVendorTaskProcessorEngine
+{
+    const OPEN_CALAIS_URL = 'https://api-eit.refinitiv.com/permid/calais';
+
+    const OPEN_CALAIS_API_KEY_METADATA_PROFILE_SYS_NAME = 'OpenCalais_PartnerData';
+    const OPEN_CALAIS_API_KEY_METADATA_FIELD_NAME = 'OpenCalaisAPIKey';
+    const OMIT_OUTPUTTING_ORIGINAL_TEXT_METADATA_FIELD_NAME = 'OmitOutputtingOriginalText';
+    const ENABLE_TICKER_EXTRACTION_METADATA_FIELD_NAME = 'EnableTickerExtraction';
+
+    const OPEN_CALAIS_MAPPING_METADATA_PROFILE_SYS_NAME = 'OpenCalais_Mapping';
+    const OPEN_CALAIS_DYNAMIC_OBJECT_MAPPING_SYSTEM_NAME = 'OpenCalais_DynamicObjectMapping';
+    const OPEN_CALAIS_MAPPING_POSTFIX = '_OpenCalais';
+
+    const SHOWTAXONOMY_SYSTEM_NAME = 'ShowTaxonomy';
+    // Mapping metadata constants
+    const RULE_NAME = 'Rule';
+    const KALTURA_FIELD_NAME_XPATH = '/metadata/Rule/Kaltura/ShowTaxonomyElement';
+
+    const OPEN_CALAIS_TOO_MANY_REQUESTS = 429;
+
+    /**
+     * @var array
+     */
+    protected $targetAllMetadataFields;
+
+    /**
+     * @var array
+     */
+    protected $targetSettableMetadataFields;
+
+    /**
+     * @var KalturaMetadataClientPlugin
+     */
+    protected $metadataPlugin;
+
+    public function __construct()
+    {
+        parent::__construct();
+        $this->metadataPlugin = KalturaMetadataClientPlugin::get(KBatchBase::$kClient);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    function handleTask(KalturaEntryVendorTask $vendorTask)
+    {
+        //Impersonate the target partner
+        KBatchBase::impersonate($vendorTask->partnerId);
```

Comment:
Once task is done, must call unimpersonate,

Reply:
@MosheMaorKaltura the endTaskOnSuccess and endTaskOnError function called already take care of that.